src/pubsub/ua_pubsub_writer.c

```diff
@@ -1994,8 +2024,10 @@ UA_WriterGroup_publishCallback(UA_Server *server, UA_WriterGroup *writerGroup) {
     UA_LOG_DEBUG(&server->config.logger, UA_LOGCATEGORY_SERVER, "Publish Callback");
 
     if(!writerGroup) {
-        UA_LOG_WARNING(&server->config.logger, UA_LOGCATEGORY_SERVER,
+        UA_LOG_ERROR(&server->config.logger, UA_LOGCATEGORY_SERVER,
                        "Publish failed. WriterGroup not found");
+        /* TODO: we can't set the state of the specific writerGroup to error, because the pointer is invalid ...
+            we would have to set the connection state to error */
```

Comment:
You're right, but this function is internal and the case writerGroup is NULL should never happen.

Reply:
ok, I have removed the comment at this PR: https://github.com/open62541/open62541/pull/3899